RNIronSourceInterstitials.js

```diff
@@ -0,0 +1,57 @@
+import { NativeModules, NativeEventEmitter } from 'react-native';
+
+const RNIronSourceInterstitials = NativeModules.RNIronSourceInterstitials;
+const IronSourceInterstialsEventEmitter = new NativeEventEmitter(RNIronSourceInterstitials);
+
+const eventHandlers = {
+  interstitialDidLoad: new Map(),
+  interstitialDidShow: new Map(),
+  interstitialDidFailToShowWithError: new Map(),
+  didClickInterstitial: new Map(),
+  interstitialDidClose: new Map(),
+  interstitialDidOpen: new Map(),
+  interstitialDidFailToLoadWithError: new Map()
+};
+
+const addEventListener = (type, handler) => {
+  switch (type) {
```

Comment:
Let's remove the switch statement here. With the previous change, this can just become:
```
const handlers = eventHandlers[type];
if (!handlers) {
  console.warn(`Event with type ${type} does not exist.`);
  return;
}

if (handlers.has(handler)) {
  console.warn(`Event with type ${type} and handler has already been added.`);
  return;
}

handlers.set(handler, IronSourceInterstialsEventEmitter.addListener(type, handler));
```
This also adds the check in to make sure you're not adding the same handler twice, as doing so would overwrite the previous stored subscription and cause a memory leak.